### PR TITLE
LG-3959 Combobox value change

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes
🎟 _Jira ticket:_ https://jira.mongodb.org/browse/LG-3959
RFC: https://docs.google.com/document/d/1LFAw9sqVyOqPKYkQ2NKDPqQLUBMJDsG-6VhIj-mw2kU/edit?usp=sharing

Combobox `onChange` callback now receives an optional 2nd argument. Use this argument to determine what value was inserted or deleted from a multiselect value.

Example:
```tsx
<Combobox 
  multiselect
  value={['apple', 'banana']}
  onChange={(val, diff) => {
    console.log(value) // ['apple']
    console.log(diff); // { diffType: 'delete', value: 'banana' }
  }} 
/>
```

```ts
interface DiffObject {
  diffType: 'insert' | 'delete';
  value: string | Array<string>;
}
```

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

**Comment by TheSonOfThomp** [comment]
We considered adding a second callback (i.e. `onValueRemoved`), however this has a few drawbacks, namely:
1. Adds the need to handle 2 callbacks
2. Doesn't handle additions
3. Awkward prop name

File: .changeset/ten-pandas-play.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  value={['apple', 'banana']}
+  onChange={(val, diff) => {
+    console.log(value) // ['apple']
+    console.log(diff); // { diffType: 'delete', value: 'banana' }
+  }} 
+/>
+```
+
+```ts
+interface DiffObject {
```
Comment (bruugey) [inline .changeset/ten-pandas-play.md:20]: nice

File: packages/combobox/src/Combobox/Combobox.spec.tsx [hunk 2]
```diff
@@ -541,6 +541,31 @@ describe('packages/combobox', () => {
...
+        const { optionElements } = openMenu();
+        expect(optionElements).not.toBeUndefined();
+        const option3 = (optionElements as HTMLCollectionOf<HTMLLIElement>)[2];
+        act(() => {
+          userEvent.click(option3);
+        });
+
+        if (select === 'multiple') {
+          expect(onChange).toHaveBeenCalledWith(
+            expect.arrayContaining(['carrot']),
```
Comment (bruugey) [inline packages/combobox/src/Combobox/Combobox.spec.tsx:558]: is this correct - an array containing an array with the string 'carrot'?
I'd expect `arrayContaining('carrot')`

File: packages/combobox/src/Combobox/Combobox.tsx [hunk 3]
```diff
@@ -224,6 +226,11 @@ export function Combobox<M extends boolean>({
       if (isMultiselect(selection)) {
         // We know M is true here
         const newSelection: SelectValueType<true> = clone(selection);
+        const multiselectOnChange = onChange as onChangeType<true>;
+        const diff: DiffObject = {
+          diffType: 'delete',
+          value: value ?? selection,
```
Comment (bruugey) [inline packages/combobox/src/Combobox/Combobox.tsx:232]: should this be `newSelection`

File: packages/combobox/src/Combobox/Combobox.spec.tsx (hunk 2 above)
Comment (TheSonOfThomp) [reply packages/combobox/src/Combobox/Combobox.spec.tsx:558]: yeah it's confusing. Its essentially "array containing these elements: [...]"

File: .changeset/ten-pandas-play.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+'@leafygreen-ui/combobox': minor
+---
+
+Combobox `onChange` callback now receives a 2nd argument. Use this argument to determine what value was inserted or deleted from a multiselect value.
```
Comment (shaneeza) [inline .changeset/ten-pandas-play.md:5]: Do you wanna include the ticket?

**Review by bruugey — APPROVED** [review]

**Pull request merged**